src/Shared/OpportunisticIntern.cs

```diff
@@ -886,6 +886,28 @@ internal void ReportStatistics(string heading)
                             return true;
                         }
                     }
+                    // see Microsoft.Build.BackEnd.BuildRequestConfiguration.CreateUniqueGlobalProperty
+                    else if (length > 15 &&
+                             candidate[0] == 'P' &&
+                             candidate[1] == 'r' &&
+                             candidate[2] == 'o' &&
+                             candidate[3] == 'j' &&
+                             candidate[4] == 'e' &&
+                             candidate[5] == 'c' &&
+                             candidate[6] == 't' &&
+                             candidate[7] == 'I' &&
+                             candidate[8] == 'n' &&
+                             candidate[9] == 's' &&
+                             candidate[10] == 't' &&
+                             candidate[11] == 'a' &&
+                             candidate[12] == 'n' &&
+                             candidate[13] == 'c' &&
+                             candidate[14] == 'e'
```

Comment:
True, but I can iterate over the constant. The compiler hopefully will unroll it a sufficient number of times to actually remove the loop entirely.